### PR TITLE
Rm all legacy

### LINKED ISSUE #1745: Remove "legacy code"
The recently released [v0.20.0](https://github.com/theupdateframework/python-tuf/releases/tag/v0.20.0) was the last release to include the legacy implementation. In [preparation for v1.0.0](https://github.com/theupdateframework/python-tuf/blob/v0.20.0/docs/1.0.0-ANNOUNCEMENT.md) it is time to actually say farewell to the legacy implementation. Here is a (likely non-exhaustive) list of related tasks:

- Identify and remove legacy code, tests and documentation
- Review and update continued resources as necessary
   - [README](https://github.com/theupdateframework/python-tuf/blob/develop/README.md) (e.g. 1.0.0 release and legacy deprecation announcements) 
   - some [python-tuf/docs/](https://github.com/theupdateframework/python-tuf/tree/develop/docs)
   - [theupdateframework.io](https://theupdateframework.io/)
   - [TUF specification](https://github.com/theupdateframework/specification)
   - 'passing', 'silver' and 'gold' criteria on [bestpractices.coreinfrastructure.org](https://bestpractices.coreinfrastructure.org/en/projects/1351)
   -  ...
- Consider renames
  - `ngclient` -> `client`
     *IMO the name `ngclient` makes most sense in relation to the "old" client and so will lose its meaning after the old client has gone, at least after a while* 
  - `api.metadata` -> `metadata`
     *IIRC the api directory was a bit of a crutch to develop new code alongside legacy code without mixing them up, which we won't need after removing the legacy code. IMO api should be designated by other means, e.g. on RTD*

### PR DESCRIPTION
**Pull request by lukpueh**
Fixes #1745 

- Removes legacy code, tests and docs
- Updates remaining docs and some project metadata accordingly

**Preliminary work**
  - https://github.com/python/peps/pull/2240
  - https://github.com/theupdateframework/theupdateframework.io/pull/31
  - https://github.com/theupdateframework/python-tuf/pull/1773
  - https://github.com/theupdateframework/python-tuf/pull/1769
  - https://github.com/theupdateframework/python-tuf/pull/1768
  - https://github.com/theupdateframework/python-tuf/pull/1767
 - Updated references in [CII Badge App](https://bestpractices.coreinfrastructure.org/de/projects/1351)
 - Checked specification for stale references to python-tuf (found none)

**Note to reviewers**:
- Excluding deleted files from diff should make review easier
- I'd appreciate a quick look through the remaining files in the repo, if I missed anything.
- Check commit messages for details

**Note of appreciation**
Many thanks to everyone who helped python-tuf to evolve! ❤️ 

Most notably (based on prior [Acknowledgements](https://github.com/theupdateframework/python-tuf/blame/b2704a56a8c5023ffd7d377d8a8e72fb26a46a8a/README.md#L130-L132) and *git shortlog* and in alphabetical order):

*Sebastien Awwad, Justin Cappos, Geremy Condra, Vladimir Diaz, Zane Fisher, Pankhuri Goyal, Jussi Kukkonen, Trishank Kuppusamy, Joshua Lock, Marina Moore, Justin Samuel, Teodora Sechkova, Tian Tian, Santiago Torres-Arias, Martin Vrachev, Yuyu Zheng, Kairo de Araujo*

### CODE REVIEW

**Comment by lukpueh** [comment]
Just moved out test updates for unittest_toolbox removal in this PR to #1792 to make reviewing easier. That PR also fixes the failing windows run seen in this PR.

**Review by jku — COMMENTED** [review]
Looks great :grin:  I might have to take another look though, it's hard to cover everything in one go.

Some observations:
* files in examples/repo_example/ reference repository_tool and repository_lib: they could be updated to indicate that those have now been deprecated
* ATTACKS.md would indeed be nice to have but that's clearly a project of its own and probably won't look much like the old one so I'm fine with leaving that for now

**Comment by lukpueh** [comment]
> files in examples/repo_example/ reference repository_tool and repository_lib: they could be updated to indicate that those have now been deprecated

Thanks for catching! This was on my mind at some point.. but then it wasn't. :D  Will push another commit here.

**Review by jku — APPROVED** [review]
looks good to me. The only additional potential removal I found was the outdated AUTHORS.txt but that does not need to be in this PR.

EDIT: and the test data generation issue I just filed #1806

**Comment by jku** [comment]
```
$ tox -e py
...
Ran 128 tests in 2.058s
```
this is so good.

(if anyone is wondering why there are so many fewer tests: we're using a lot more subtests and those are not counted individually: my Stetson-Harrison assessment is that the new code on average has better coverage than the old one did, although specific areas may be worse).

File: tox.ini
Comment (MVrachev) [inline tox.ini:41]: Did you check if all of the tools recursively check the next folders in `tuf/` - `tuf/ngclient` and `tuf/api`?

**Review by MVrachev — APPROVED** [review]
I was waiting for this pr for a long time!
Great stuff here.

I totally forgot that the old updater is 3070 lines, repository_lib 2306 lines, and repository_tool 3291... wow

Also, thanks for the last commitment! I was planning on doing that.

Only one question: do we have a test for key revocation?

**Comment by jku** [comment]
> Only one question: do we have a test for key revocation?

In the sense that the old tests used the word "revocation", yes we do.

test_updater_key_rotations.py contains loads of different cases of key rotations: some remove ("revoke") old keys, some don't. test_updater_top_level_update.py has tests for fast forward recovery which include complete key rotations (these are made easier to read in #1766).

**Comment by jku** [comment]
I should have stated this explicitly: 
* I think the PR is good and we should deal with any fallout with new issues after merging
* I did not merge myself as I wanted other maintainers to have time to review as it's a major change

**Comment by lukpueh** [comment]
Thanks for your reviews, @jku and @MVrachev!


> * files in examples/repo_example/ reference repository_tool and repository_lib: they could be updated to indicate that those have now been deprecated

Fixed in 9816c40.

> Did you check if all of the tools recursively check the next folders in `tuf/` - `tuf/ngclient` and `tuf/api`?

Yes, I did. And yes, they do.

> Only one question: do we have a test for key revocation?

I admit that I didn't question test completeness. I just `rm tests/test_*_old.py` (relying on 22fe1e69e4315ce3e96d712fbf8d50871ce5f64f) and cross-checked with effective test coverage, which didn't decrease.

**Comment by lukpueh** [comment]
> I should have stated this explicitly:
> 
> * I think the PR is good and we should deal with any fallout with new issues after merging
> * I did not merge myself as I wanted other maintainers to have time to review as it's a major change

@-'other (active) maintainers', i.e. @mnm678, @joshuagl, mind wrapping this up? 🚀 😃

**Review by mnm678 — APPROVED** [review]
Thanks to all the people who worked on the new client! This is really exciting, and looks good to me.

**Comment by lukpueh** [comment]
Thanks for the reviews, everyone! Merging...

**Pull request merged**
